### ISSUE TITLE
page.url not giving correct url on application side redirect?

### ISSUE THREAD

**Issue by nabati** [puppeteer/puppeteer#1370]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version:
0.12
* Platform / OS version:
Mac OSX
* URLs (if applicable):

**What steps will reproduce the problem?**
Not sure where the issue is but, I was expecting these to return the same values;

```
await page.url() 
``` 

and 
```
await page.evaluate(() => window.location.href)
```

but they do not. `page.url()` seems to only reflect the last url that was explicitly navigated to via one of puppeteers helper mehtod, but does not seem to get updated if for instance the navigation happens because of a user event (programmatically within a web app). 

`page.evaluate` seems to reflect the true url of the current navigation state.

**Comment by blue-cp** [comment]
we are seeing the same issues. on our side. This is still an issue on 0.13.0
Sample Urls:

https://t.co/eo4xEnfO91
https://t.co/eo4xEnfO91
https://t.co/redirect?url=http%3A%2F%2Ffacebook.com%2Fmetinucaile%3Ft%3D1%26cn%3DZmxleGlibGVfcmVjc18y%26refsrc%3Demail%26iid%3D3e3a9eabb3fa4a7d941314810094c968%26uid%3D298104373%26nid%3D244%2B276959240&amp;t=1+1510582428261&amp;cn=ZmxleGlibGVfcmVjc18y&amp;sig=9760ebf9e854b91dfa04ca7ad0212fef20d01459&amp;iid=3e3a9eabb3fa4a7d941314810094c968&amp;uid=298104373&amp;nid=244+276959240

**Comment by aslushnikov** [comment]
This will be fixed as soon as #257 is addressed.

Closing in favor of #257.
